### PR TITLE
Add CD to build and publish extension to Chrome & Firefox stores

### PR DESCRIPTION
**Pull request by djahandarie**
(Stacked PR on #91)

This sets up the following workflows:

* `create-prerelease-on-tag`, which runs when a `*.*.*.*` tag is created:
  * runs `dev/build.sh` to generate all the zips (chrome, chrome-dev, firefox, firefox-dev),
  * creates a GitHub release (marked as prerelease) with the zips, and an automatically generated changelog using the settings in `.github/release.yml`
* `create-firefox-development`, which runs when a prerelease is created:
  * submits the firefox-dev zip to Mozilla for signing,
  * adds the signed xpi to the prerelease (so people can download/install it)
  * adds the download URL for the signed xpi in the prerelease into `update.json` on the `metadata` branch, which will cause anyone who already has the Firefox development version to auto-update within 24hrs
* `create-chrome-development`, which runs when a prerelease is created:
  * submits the chrome-dev zip to the Chrome Web Store to be published under "Yomitan (development build)" (retries when failed)
* `create-firefox`, which runs when the release is switched from prerelease to release:
  * submits the firefox zip to Firefox Addons to be published under "Yomitan"
* `create-chrome` which runs when the release is switched from prerelease to release:
  * submits the chrome zip to Chrome Web Store to be published under "Yomitan"

The idea is that first we'd create a tag, which would cause the first 3 actions to run, have it tested, and then once it is ready, simply just change the release from prerelease to a normal release which would cause the last 2 actions to run.

### CODE REVIEW

File: .github/workflows/create-prerelease-on-tag.yml
Comment (Rudo2204) [inline .github/workflows/create-prerelease-on-tag.yml:20]: Curious why you don't use `actions/setup-node@v3` here

File: .github/workflows/create-prerelease-on-tag.yml
Comment (Rudo2204) [inline .github/workflows/create-prerelease-on-tag.yml:32]: Usually I like to add `draft: true` here so we can add a changelog to the release before publishing it

File: .github/workflows/publish-chrome-development.yml
Comment (Rudo2204) [inline .github/workflows/publish-chrome-development.yml:38]: I wonder if you even need to use `parseInt` here if `attemptNumber` and `maxAttempts` were set to be integer already in the `workflow_dispatch` inputs

File: .github/workflows/publish-chrome-development.yml
Comment (Rudo2204) [inline .github/workflows/publish-chrome-development.yml:14]: 10 max attempts seem excessive tbh

File: .github/workflows/publish-firefox-development.yml
Comment (Rudo2204) [inline .github/workflows/publish-firefox-development.yml:83]: I wonder if we could do better by pulling a list of changes from the commit history here instead and create a better commit message

File: .github/workflows/create-prerelease-on-tag.yml
Comment (djahandarie) [reply .github/workflows/create-prerelease-on-tag.yml:20]: Because I copied it from the existing CI, but I will update both occurrences, thanks :+1:

File: .github/workflows/create-prerelease-on-tag.yml
Comment (djahandarie) [reply .github/workflows/create-prerelease-on-tag.yml:32]: Hmm:
* the changelog is going to be auto-generated thanks to `generate_release_notes` / `release.yml`
* since the `publish-firefox-development` workflow will immediately run and cause everyone with the dev version to get the update, I think it's better that it's not a draft at this point but a prerelease
* is it not possible to modify the description afterwards if there's something we don't like about the auto-generated version?

File: .github/workflows/publish-chrome-development.yml
Comment (djahandarie) [reply .github/workflows/publish-chrome-development.yml:38]: Not sure, but I stole this from another workflow so there's probably a reason it's like this. If I figure out how to test this I can try removing them.

File: .github/workflows/publish-chrome-development.yml
Comment (djahandarie) [reply .github/workflows/publish-chrome-development.yml:14]: The reason it's this long is because it's a matter of waiting for the the previous version to be approved on the Web Store, which can take days. In particular it's 12 hrs delay * 10 times, so 5 days total, which I think is reasonable for this.

File: .github/workflows/publish-firefox-development.yml
Comment (djahandarie) [reply .github/workflows/publish-firefox-development.yml:83]: Hmm, you mean to leave like a changelog in the commit? We could pull it from the release description. But I'm not sure it's necessary since normally no one would look at the branch lol. Maybe just the verison number and/or a link to the release or something

Comment (djahandarie) [reply .github/workflows/publish-firefox-development.yml]: Gave the version number + link idea a shot, let me know what you think. https://github.com/themoeway/yomitan/pull/92/commits/765e229965838dfcb26f3afeef950aeba22a5d3d#diff-8d5cfe2441357d30ded1a905f29730cd09ae031e3c8e850327fea566ae2b4109R78

File: .github/workflows/create-prerelease-on-tag.yml
Comment (Rudo2204) [reply .github/workflows/create-prerelease-on-tag.yml:32]: That seems fine to me

File: .github/workflows/publish-chrome-development.yml
Comment (Rudo2204) [reply .github/workflows/publish-chrome-development.yml:14]: Ah it has to wait for approval, I didn't know this. Thanks!

File: .github/workflows/publish-firefox-development.yml
Comment (Rudo2204) [reply .github/workflows/publish-firefox-development.yml]: Yeah this looks better than the generic "Update" message for sure

File: .github/workflows/publish-chrome-development.yml
Comment (djahandarie) [reply .github/workflows/publish-chrome-development.yml:38]: Turns out that it is necessary. All the env inputs are strings from the perspective of `js-eval-action`.

**Review by Timm04 — DISMISSED** [review]

**Review by Rudo2204 — DISMISSED** [review]

**Review by Timm04 — DISMISSED** [review]

**Pull request merged**
